### PR TITLE
Refactored footer icons + many new icons

### PR DESCRIPTION
**Pull request by danisztls**
+ sorted icons alphabetically
+ social icons for Mastodon, Matrix and Telegram
+ add rel="me" to social links

rel="me" is a de facto standard for this kind of links 
check: https://microformats.org/wiki/rel-me

FontAwesome is missing an icon for Matrix. Request for inclusion was opened in 2018 but I guess they don't want to add an icon without Matrix.org paying the comission.

### CODE REVIEW

File: _includes/social/icon_partial.html [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {% if include.isDisplayed %}
 <li>
     <a href="{{ include.url }}"
-       title="{{ site.data.language.str_follow_on }} {{ include.social }}">
+       title="{{ site.data.language.str_follow_on }} {{ include.social }}"
+       rel="me">
```
Comment (sylhare) [inline _includes/social/icon_partial.html:5]: what's that `rel="me"` attribute?

File: _includes/social/icons.md [hunk 2]
```diff
@@ -86,9 +98,9 @@
 {% assign tb_on = site.data.social.tumblr or site.theme_settings.tumblr %}
 {% include social/icon_partial.html isDisplayed=tb_on url=tb_url social="Tumblr" %}
 
-{% assign gl_url = "https://gitlab.com/" | append: site.data.social.gitlab %}
-{% assign gl_on = site.data.social.gitlab or site.theme_settings.gitlab %}
-{% include social/icon_partial.html isDisplayed=gl_on url=gl_url social="Gitlab" %}
+{% assign tg_url = "https://t.me/@" | append: site.data.social.telegram %}
+{% assign tg_on = site.data.social.telegram or site.theme_settings.telegram %}
+{% include social/icon_partial.html isDisplayed=tg_on url=tg_url social="Telegram" %}
```
Comment (sylhare) [inline _includes/social/icons.md:103]: Did you test that the icon exist for Telegram, Mastodon and Matrix?

Comment (sylhare) [reply _includes/social/icons.md:103]: Oh ok just saw your comment in the PR.
If there are no icon for Matrix then it should not be in the there.

File: _includes/social/icon_partial.html (hunk 1 above)
Comment (sylhare) [reply _includes/social/icon_partial.html:5]: Ok saw the link

Comment (danisztls) [reply _includes/social/icon_partial.html:5]: Mastodon read. It is used to verify authenticity of link.

**Comment by danisztls** [comment]
I had refactored the footer icons, which will break configuration, but make icons easier to add and modify besides making the code cleaner with all the data on `_data`.

One option to improve it further is to move user data to `_config,yml` separating it from the icon 'templates' data.

Something like this:

***_includes/social/icons.yml****
``ỳml
github:
  pre: "https://github.com/"
  pos:
  icon:
```

***_config.yml***
``ỳml
me:
  github: "danisztls"
```

**Comment by danisztls** [comment]
Added a bunch of low hanging fruits.

- Calendly
- Google Scholar
- Map (OpenStreetMap)
- Medium
- Patreon
- Phone
- SMS
- Twitch
- VK

File: _includes/social/icons.md (hunk 2 above)
Comment (sylhare) [reply _includes/social/icons.md]: Well I'd rather stay with Font Awesome for now.

File: _data/icons.yml [hunk 3]
```diff
@@ -0,0 +1,177 @@
+# Icons on the footer
+# Usage: Modify 'id' for the icon to show up.
+behance:
+  pre: "https://www.behance.net/" 
+  id: "sam" # e.g. "sam"
+  pos:
+  icon:
```
Comment (sylhare) [inline _data/icons.yml:7]: Nah that is a pain to add as an icon,
I'd rather have it done with the code

Comment (danisztls) [reply _data/icons.yml:7]: With the change I proposed above (but did not implement) the user would only have to put the desired icons on a yaml collection on a `_config`. Similar thing could be made to 'share icons'.

Comment (danisztls) [reply _data/icons.yml:7]: done

**Comment by coder-lg** [comment]
What is Matrix mentioned in the first comment?

**Comment by danisztls** [comment]
> What is Matrix mentioned in the first comment?

It is a decentralized real-time chat protocol. It has many clients like Element and Fluffy.  Although it is no so decentralized yet as most users and servers are located on Matrix.org namespace.

**Comment by coder-lg** [comment]
So, is it a chat app server?

File: _config.yml [hunk 4]
```diff
@@ -30,6 +30,11 @@ disqus_shortname:                                       # Your discus shortname
 cusdis_app_id:                                          # Your cusdis data-app-id
 color_theme: auto                                       # auto, dark or light
 
+# Social icons
+me:
+  github: "sylhare"
+  stack-overflow: "7747942/sylhare"
```
Comment (sylhare) [inline _config.yml:36]: Why is there social icons in the `_config.yaml` and the `_data/icons.yml`?

Let's have like before a `social.yaml` where you put your username (github, stackoverflow, ...) and the `icons.yaml` could be just the boilerplate to actually build the icons

File: _data/icons.yml [hunk 5]
```diff
@@ -0,0 +1,141 @@
+# Icons on the footer
+behance: # e.g. "sam"
+  pre: "https://www.behance.net/" 
+  pos:
+  icon:
```
Comment (sylhare) [inline _data/icons.yml:5]: Do you think we can remove the ones that are empty?

Comment (danisztls) [reply _data/icons.yml:5]: Yes.

**Comment by danisztls** [comment]
> So, is it a chat app server?

Protocol. Has many clients, servers/bridges.

**Comment by coder-lg** [comment]
So, why do you need to add Matrix? You could possibly add Element rather than Matrix, right?

**Comment by danisztls** [comment]
> So, why do you need to add Matrix? You could possibly add Element rather than Matrix, right?

I can use my account with any client and connect to networks like IRC, Telegram and Discord giving that they have bridges. That is my Matrix address on matrix.org. Element is just one client.

**Comment by coder-lg** [comment]
Well, there's good **AND** bad in it. Goodis that people can contact you on bug info's directly instead of issues. Bad is that, there might be spam messages.

**Comment by coder-lg** [comment]
And spam is irritating.

**Comment by coder-lg** [comment]
And I am now interested in Element

**Comment by coder-lg** [comment]
Well, Matrix is new here.

**Comment by coder-lg** [comment]
I got an idea. Why not you get Matrix's icon, customize it and use it here?

**Comment by danisztls** [comment]
> I got an idea. Why not you get Matrix's icon, customize it and use it here?

Adding exceptions make the code less maintainable and it's not critical to have. I used a close substitute.

**Comment by coder-lg** [comment]
Ok

**Review by sylhare — COMMENTED** [review]
From what I see, it will be a breaking change since you'll have to update one yaml or another

**Comment by danisztls** [comment]
> From what I see, it will be a breaking change since you'll have to update one yaml or another

Yes

**Comment by sylhare** [comment]
It is now merged, I made some change to make it backward compatible 👍 
Thanks for the PR @danisztls that's a whole lot of new icons we have now ehehe

**Pull request merged**